### PR TITLE
Add gravatar system with name initials fallback

### PR DESCRIPTION
**Pull request by alireza-ahmadi**
Thanks to @bengott's amazing library, I have implemented avatar(Gravatar with two character initials fallback) system for Libreboard, Hope it works.
avatar

### CODE REVIEW

File: client/lib/utils.js [hunk 1]
```diff
@@ -142,6 +142,14 @@ Utils = {
                 return 'labels.' + labels[labelId] + '.' + key;
             }
         }
+    },
+
+    // Extract userId from user object
+    getUserId: function(user) {
```
Comment (mquandalle) [inline client/lib/utils.js]: I think we don't want this function. In the code we should always know if a variable `x` is a user object (in this case we just use `x._id`) or an id (`x`). This function can bring confusion during the development.

File: client/views/users/helpers.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
     memberType: function() {
-        var user = Users.findOne(this.memberId);
+        var user = Users.findOne(Utils.getUserId(this.user));
         return user && user.isBoardAdmin() ? 'admin' : 'normal';
     }
 });
+
+Avatar.options = {
+    fallbackType: 'initials'
+};
```
Comment (mquandalle) [inline client/views/users/helpers.js]: Please add a new line at the end of the file.

File: client/views/users/templates.html [hunk 3]
```diff
@@ -185,12 +178,15 @@ <h3 class="bottom" style="margin-right: 40px;">
...
+        {{> avatar user=userData size=size }}
+        {{#if showStatus }}
+            <span class="member-status {{# if userData.profile.status}}active{{/if}}"></span>
+        {{/if }}
+        {{#if showMemberType}}
+            <span class="member-type {{ memberType }}"></span>
+        {{/if }}
     </div>
-</template>
+</template>
```
Comment (mquandalle) [inline client/views/users/templates.html]: new line

File: client/views/users/templates.html [hunk 4]
```diff
@@ -185,12 +178,15 @@ <h3 class="bottom" style="margin-right: 40px;">
...
-        <span class="member-status {{# if user.profile.status}}active{{/if}}"></span>
-        <span class="member-type {{ memberType }}"></span>
+<template name="userAvatar">
+    <div class="member {{size}} {{ class }} {{# if draggable }}js-member{{else}}js-member-on-card-menu{{/ if}}"
+         title="{{ userData.profile.name}} ({{ userData.username}})">
+        {{> avatar user=userData size=size }}
+        {{#if showStatus }}
+            <span class="member-status {{# if userData.profile.status}}active{{/if}}"></span>
+        {{/if }}
+        {{#if showMemberType}}
```
Comment (mquandalle) [inline client/views/users/templates.html]: It seems that every time you call `{{> userAvatar }}` with the `showStatus` option, you also use the `showMemberType`. Maybe we could have only one option that would display the two icons. (:+1: for the option BTW)

File: client/views/widgets/events.js [hunk 5]
```diff
@@ -45,7 +45,7 @@ var getMemberIndex = function(board, searchId) {
 Template.memberPopup.events({
     'click .js-change-role': function(event, t) {
         var currentBoard = Boards.findOne();
-        var memberIndex = getMemberIndex(currentBoard, this.memberId);
+        var memberIndex = getMemberIndex(currentBoard, Utils.getUserId(this.user));
```
Comment (mquandalle) [inline client/views/widgets/events.js]: I believe this change is not desirable.

**Comment by mquandalle** [comment]
Hello @alireza-ahmadi,

Thank you for your great work. Please consider my inline comments before I can merge this PR.

File: client/views/users/templates.html [hunk 6]
```diff
@@ -81,10 +79,7 @@ <h1 id="landingLogo">
         <div class="tabbed-pane-header">
             <div class="tabbed-pane-header-wrapper clearfix">
                 <a class="tabbed-pane-header-image profile-image ed js-change-avatar-profile" href="#">
-                    <span class="profile-image-initials">{{ firstChar profile.profile.name }}</span>
-                    {{#if isTrue currentUser.username profile.username }}
-                        <span class="change-text">{{_ 'change-avatar'}}</span>
-                    {{ /if }}
+                    {{> userAvatar user=profile size="large" removeWrapper=true}}
```
Comment (mquandalle) [inline client/views/users/templates.html]: I think the name `noWrap` is better than `removeWrapper`.

File: client/lib/utils.js (hunk 1 above)
Comment (alireza-ahmadi) [inline client/lib/utils.js]: I have removed this method. Now developers can obtain user ID using `this.user._id`. Also in last commit I have replaced `this.user._id` with `this.userId` to make it more similar to original code, If you are OK with `this.user._id`, you can ignore last commit. (Sorry for my weak English grammar ;))

File: client/views/users/templates.html (hunk 4 above)
Comment (alireza-ahmadi) [inline client/views/users/templates.html]: Two options replaced with single `showBadges` options.

**Comment by alireza-ahmadi** [comment]
I have added the commit email to my Github account, Also I have fixed items that you mentioned in the comments. Please let me know if there is any other problems with this pull request.

File: client/lib/utils.js (hunk 1 above)
Comment (alireza-ahmadi) [inline client/lib/utils.js]: I have removed this method. Now developers can obtain user ID using `this.user._id`. Also in last commit I have replaced `this.user._id` with `this.userId` to make it more similar to original code, If you are OK with `this.user._id`, you can ignore last commit. (Sorry for my weak English grammar ;))

**Comment by mquandalle** [comment]
Thank you very much. I merged your changes in df7e69b3c41d.

File: client/views/users/member.styl [hunk 7]
```diff
@@ -16,15 +16,72 @@
...
     margin: 0 4px 4px 0;
     overflow: hidden;
     position: relative;
-    width: 30px;
+    width: 50px;
     cursor: pointer;
     user-select: none;
     z-index: 0;
     text-decoration: none;
+
```
Comment (mquandalle) [inline client/views/users/member.styl]: BTW there was some trailing whitespaces here; we try to avoid them :-)

Comment (alireza-ahmadi) [inline client/views/users/member.styl]: Thanks, I will consider this point in future contributions.

**Pull request closed without merging**